beacon-chain/state/getters.go

```diff
@@ -13,36 +13,54 @@ import (
 // EffectiveBalance returns the effective balance of the
 // read only validator.
 func (v *ReadOnlyValidator) EffectiveBalance() uint64 {
+	if v == nil || v.validator == nil {
+		return 0
+	}
 	return v.validator.EffectiveBalance
 }
 
 // ActivationEligibilityEpoch returns the activation eligibility epoch of the
 // read only validator.
 func (v *ReadOnlyValidator) ActivationEligibilityEpoch() uint64 {
+	if v == nil || v.validator == nil {
+		return 0
```

Comment:
given 0 is a valid epoch but im sure if this is a sound approach, return max int?

Reply:
should i add error also here?

Reply:
returned FarFutureEpoch